ext/session/session.c

```diff
@@ -664,24 +664,41 @@ static PHP_INI_MH(OnUpdateName) /* {{{ */
 	SESSION_CHECK_ACTIVE_STATE;
 	SESSION_CHECK_OUTPUT_STATE;
 
-	/* Numeric session.name won't work at all */
-	if ((!ZSTR_LEN(new_value) || is_numeric_string(ZSTR_VAL(new_value), ZSTR_LEN(new_value), NULL, NULL, 0))) {
-		int err_type;
+	int err_type;
 
-		if (stage == ZEND_INI_STAGE_RUNTIME || stage == ZEND_INI_STAGE_ACTIVATE || stage == ZEND_INI_STAGE_STARTUP) {
-			err_type = E_WARNING;
-		} else {
-			err_type = E_ERROR;
-		}
+	if (stage == ZEND_INI_STAGE_RUNTIME || stage == ZEND_INI_STAGE_ACTIVATE || stage == ZEND_INI_STAGE_STARTUP) {
+		err_type = E_WARNING;
+	} else {
+		err_type = E_ERROR;
+	}
 
+	if (ZSTR_LEN(new_value) == 0) {
+		/* Do not output error when restoring ini options. */
+		if (stage != ZEND_INI_STAGE_DEACTIVATE) {
+			php_error_docref(NULL, err_type, "session.name \"%s\" cannot be empty", ZSTR_VAL(new_value));
+		}
+		return FAILURE;
+	}
+	/* Nul bytes are not allowed */
+	if (ZSTR_LEN(new_value) != strlen(ZSTR_VAL(new_value))) {
```

Comment:
This check is highly non-obvious. I was about to suggest changing the second to `ZSTR_LEN` as well.

Reply:
```suggestion
	if (CHECK_NULL_PATH(ZSTR_VAL(new_value), ZSTR_LEN(new_value))) {
```

Maybe it makes sense to introduce another macro, which expects a `zend_string`.